http/server.ts

```diff
@@ -1,111 +1,117 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
-import { listen, Conn, toAsyncIterator, Reader, Writer, copy } from "deno";
-import { BufReader, BufState, BufWriter } from "../io/bufio.ts";
+
+import { Conn, copy, listen, Reader, toAsyncIterator, Writer } from "deno";
+import { BufReader, BufWriter } from "../io/bufio.ts";
 import { TextProtoReader } from "../textproto/mod.ts";
 import { STATUS_TEXT } from "./http_status.ts";
 import { assert } from "../testing/mod.ts";
+import { defer, Deferred } from "../async/deferred.ts";
+import { Key, pathToRegexp } from "./path_to_regexp.ts";
+import { BodyReader, ChunkedBodyReader } from "./readers.ts";
 
-interface Deferred {
-  promise: Promise<{}>;
-  resolve: () => void;
-  reject: () => void;
-}
+export type HttpHandler = (req: ServerRequest) => Promise<any>;
```

Comment:
I think `Promise<any>` return type is bad, since it's not type safe. Can't it be `Promise<void>` since we don't expect any return type?

Reply:
I know. but `:Promise<void>` function can't accept one-line arrow function with non-void return vlaue:
```ts
server.handle("/", req => notVoidFunction(req))
```
I don't know which is better. But I chose one not restrictive.

Reply:
How about this?
```suggestion
export type HttpHandler = (req: ServerRequest) => void | Promise<void>;
```

Reply:
what about `Promise<unknown>`  ?

Reply:
Or just 
```suggestion
export type HttpHandler = (req: ServerRequest) => unknown;
```

since TS should let you `await` something that’s not a promise.